spring-cloud-contract-tools/spring-cloud-contract-gradle-plugin/src/main/groovy/org/springframework/cloud/contract/verifier/plugin/GenerateServerTestsTask.groovy

```diff
@@ -16,99 +16,113 @@
+import org.gradle.api.file.Directory
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.TaskAction
-
-import org.springframework.cloud.contract.spec.Contract
 import org.springframework.cloud.contract.spec.ContractVerifierException
 import org.springframework.cloud.contract.verifier.TestGenerator
 import org.springframework.cloud.contract.verifier.config.ContractVerifierConfigProperties
 import org.springframework.cloud.contract.verifier.config.TestFramework
-
-import static org.springframework.cloud.contract.verifier.plugin.SpringCloudContractVerifierGradlePlugin.COPY_CONTRACTS_TASK_NAME
+import org.springframework.cloud.contract.verifier.config.TestMode
 
 /**
  * Task used to generate server side tests
  *
  * @since 1.0.0
  */
 @CompileStatic
-class GenerateServerTestsTask extends ConventionTask {
+class GenerateServerTestsTask extends DefaultTask {
+	static final String TASK_NAME = 'generateContractTests'
 
-	File generatedTestSourcesDir
+	@InputDirectory
+	Provider<Directory> contractsDslDir
+	@Input
+	Provider<String> includedContracts
+	@Input
+	@Optional
+	String nameSuffixForTests
+	@Input
+	@Optional
+	String basePackageForTests
+	@Input
+	@Optional
+	String baseClassForTests
+	@Input
+	@Optional
+	String packageWithBaseClasses
+	@Input
+	List<String> excludedFiles
+	@Input
+	List<String> ignoredFiles
+	@Input
+	List<String> includedFiles
+	@Input
+	String[] imports
+	@Input
+	String[] staticImports
+	@Input
+	TestMode testMode
+	@Input
+	TestFramework testFramework
+	@Input
+	Map<String, String> baseClassMappings
+	@Input
+	boolean assertJsonSize
+	@Input
+	boolean failOnInProgress
 
-	//TODO: How to deal with @Input*, @Output* and that domain object?
-	ContractVerifierExtension configProperties
-	GradleContractsDownloader downloader
+	@OutputDirectory
+	File generatedTestSourcesDir
+	@OutputDirectory
+	File generatedTestResourcesDir
 
 	@TaskAction
 	void generate() {
-		logger.info("Generated test sources dir [${getGeneratedTestSourcesDir()}]")
-		Task copyContractsTask = project.getTasksByName(COPY_CONTRACTS_TASK_NAME, false).first()
-		ContractVerifierConfigProperties props = props(copyContractsTask)
-		File contractsDslDir = contractsDslDir(copyContractsTask, props)
-		if (getConfigProperties().getContractDependency()) {
-			project.logger.debug("Updating the stubs locations for the case where we have a JAR with contracts")
-			props.contractsDslDir = contractsDslDir
-			props.includedContracts = ".*"
-		}
+		logger.info("Generated test sources dir [${generatedTestSourcesDir}]")
+		logger.info("Generated test resources dir [${generatedTestResourcesDir}]")
+		// TODO: Double check this. `ContractDependency` seems to always be `true`. I believe it was wrong here:
+		// if (getConfigProperties().getContractDependency()) {
```

Comment:
Need a feedback on this. Looks like it was a bug here, but I'm not 100% sure...

Reply:
Yeah it looks like it. We should check if either group id of a dependency is set or its string notation.

Reply:
@marcingrzejszczak question is: do we really need to do this here? If you check, `GradleContractsDownloader` is updating that `includedContracts` property to some calculated value in case there is group id / we downloaded it. Then this property is only used here. And if in exactly the same scenario we reset this property back to default value, then whole idea of calculating it in the `GradleContractsDownloader` is not clear to me. Moreover, maven plugin doesn't seem to have logic like this.

Reply:
Absolutely we don't need to do it here. Actually, we already do it in the `GradleContractsDownloader` as you say, so this shouldn't even be here.

Reply:
Ok, removed.